python-app/docker/action.yml

```diff
@@ -0,0 +1,61 @@
+name: Build
+description: Build and publish image
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Check if step is needed
+      run: |
+        echo docker_needed=${{ startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/feature/') || github.ref == 'main' }} >> $GITHUB_ENV
+      shell: bash
+
+    - name: Clone
+      uses: actions/checkout@v2
+      if: ${{ env.docker_needed == 'true' }}
+
+    - name: Build image name (lowercase)
+      run: |
+        echo IMAGE="ghcr.io/${GITHUB_REPOSITORY,,}" >> $GITHUB_ENV
+      shell: bash
```

Comment:
Can you move this up as it's part of the setup happening in both `docker_needed` == true/false cases